tests/end_to_end/end_to_end.cpp

```diff
@@ -592,6 +592,137 @@ TEST_CASE("bindmonitor-interpret", "[bindmonitor_interpret]")
     hook.detach();
 }
 
+TEST_CASE("bindmonitor-jit", "[bindmonitor_jit]")
+{
+    _test_helper test_helper;
+
+    ebpf_handle_t program_handle;
+    const char* error_message = NULL;
+    ebpf_handle_t map_handles[4];
+    uint32_t count_of_map_handles = 2;
+    uint64_t fake_pid = 12345;
+    uint32_t result;
+
+    program_information_provider_t bind_program_information(EBPF_PROGRAM_TYPE_BIND);
+
+    REQUIRE(
+        (result = ebpf_api_load_program(
+             SAMPLE_PATH "bindmonitor.o",
+             "bind",
+             EBPF_EXECUTION_JIT,
+             &program_handle,
+             &count_of_map_handles,
+             map_handles,
+             &error_message),
+         error_message ? printf("ebpf_api_load_program failed with %s\n", error_message) : 0,
+         ebpf_api_free_string(error_message),
+         error_message = nullptr,
+         result == EBPF_SUCCESS));
+
+    single_instance_hook_t hook;
+
+    std::string process_maps_name = "bindmonitor::process_maps";
+    std::string limit_maps_name = "bindmonitor::limits_map";
+
+    REQUIRE(
+        ebpf_api_pin_object(
+            map_handles[0],
+            reinterpret_cast<const uint8_t*>(process_maps_name.c_str()),
+            static_cast<uint32_t>(process_maps_name.size())) == EBPF_SUCCESS);
+    REQUIRE(
+        ebpf_api_pin_object(
```

Comment:
To me it looks like this test tries to do too many things that need not be coupled.
E.g., pinning maps, enumerating maps, and attaching to the bind hook are three orthogonal things that should be in separate tests.  Point is to make tests independent and granular so there's lots of granular tests instead of a few monolithic tests.

Reply:
Good suggestion. Refactored the tests to make them simpler and remove duplication.